src/Transports.AspNetCore/WebSockets/GraphQLWebSocketOptions.cs

```diff
@@ -0,0 +1,41 @@
+namespace GraphQL.Server.Transports.AspNetCore.WebSockets;
+
+/// <summary>
+/// Configuration options for a WebSocket connection.
+/// </summary>
+public class GraphQLWebSocketOptions
+{
+    /// <summary>
+    /// The amount of time to wait for a GraphQL initialization packet before the connection is closed.
+    /// The default is 10 seconds.
+    /// </summary>
+    public TimeSpan? ConnectionInitWaitTimeout { get; set; }
+
+    /// <summary>
+    /// The amount of time to wait between sending keep-alive packets.
+    /// The default is disabled.
+    /// <br/><br/>
+    /// Keep in mind that the 'subscription-transport-ws' implementation typically
+    /// disconnects clients if a keep-alive packet has not been received for 20 seconds,
+    /// when keep-alive packets are enabled, so it is recommended to keep the keep-alive
+    /// packets disabled or use a value less than 20 seconds.
+    /// </summary>
+    public TimeSpan? KeepAliveTimeout { get; set; }
+
+    /// <summary>
+    /// The amount of time to wait to attempt a graceful teardown of the WebSockets protocol.
+    /// The default is 10 seconds.
+    /// </summary>
+    public TimeSpan? DisconnectionTimeout { get; set; }
+
+    /// <summary>
+    /// Disconnects a subscription from the client if the subscription source dispatches an
+    /// <see cref="IObserver{T}.OnError(Exception)"/> event.
+    /// </summary>
+    public bool DisconnectAfterErrorEvent { get; set; } = true;
+
+    /// <summary>
+    /// Disconnects a subscription from the client there are any GraphQL errors during a subscription.
```

Comment:
```suggestion
    /// Disconnects a subscription from the client in case of any GraphQL errors during a subscription.
```

Reply:
> How it works? Is it assumed that a subscription can sometimes work even if there are errors? By the way, what errors?

It is allowed for a GraphQL subscription to periodically return data in the form of a properly formatted ExecutionResult (for the subscribed field only). Also it can return errors in the same manner. 

So let’s say that a user subscribed to a notifications about new high scores for a game. Let’s days they requested Score and Name. If both are non null fields, but the data source generating the source events gave a null value for Name, the DocumentExecuter would generate a proper ExecutionResult with an error message saying that a null value was supplied for a non null field. This result can be passed back through the WebSocket connection to the client without breaking the WebSocket connection or subscription. However, it may be desired that for some unanticipated errors, the connection is purposely dropped, forcing the client to reconnect, which may reset whatever the fault was on the server end.

Reply:
Note that all validation checks occur before the subscription is completely set up.